src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    static final Integer MIN_CAPACITY = 10;
+    private T[] array;
+    private Integer elementData = 0;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[MIN_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow(1, this.array.length);
+        this.array[elementData] = value;
+        elementData += 1;
     }
 
     @Override
     public void add(T value, int index) {
+        grow(1, this.array.length);
+        if (index < elementData) {
+            T[] cloneOfArray = this.array.clone();
+            this.array[index] = value;
+            elementData += 1;
+            System.arraycopy(cloneOfArray, index, this.array, index + 1, elementData - index - 1);
+        } else if (index == elementData) {
+            this.add(value);
+            elementData += 1;
+        } else {
+            throw new ArrayIndexOutOfBoundsException("Index: " + index + " Size: " + elementData);
+        }
+    }
 
+    private void grow(int lengthOfElements, Integer length) {
```

Comment:
you don't need to pass here anything, you have access to your array.length anyway, so you just can increase it in 1.5 time when you need to